share/eds/blockstore.go

```diff
@@ -0,0 +1,115 @@
+package eds
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/filecoin-project/dagstore"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	ipld "github.com/ipfs/go-ipld-format"
+)
+
+var _ blockstore.Blockstore = (*Blockstore)(nil)
+
+var (
+	errUnsupportedOperation = errors.New("unsupported operation")
+	errShardNotFound        = errors.New("the provided cid does not map to any shard")
+)
+
+// Blockstore implements the blockstore.Blockstore interface on an EDSStore.
+// The lru cache approach is heavily inspired by the existing implementation upstream.
+// We simplified the design to not support multiple shards per key, call GetSize directly on the
+// underlying RO blockstore, and do not throw errors on Put/PutMany. Also, we do not abstract away
+// the blockstore operations.
+//
+// The intuition here is that each CAR file is its own blockstore, so we need this top level
+// implementation to allow for the blockstore operations to be routed to the underlying stores.
+type Blockstore struct {
+	store *Store
+	cache *blockstoreCache
+}
+
+func NewEDSBlockstore(store *Store, cache *blockstoreCache) *Blockstore {
```

Comment:
No one can create a blockstoreCache so this is effectively private. Also, I don't see a use case for this outside of the pkg and suggest privatizing. The struct and its constructor